### DISCUSSION TITLE
Opinion on 1D FFT parallelization

### DISCUSSION THREAD

**Discussion by arunkv7** [FourierFlows/FourierFlows.jl#346] [General]
Hi, 
I am trying to see how to make the one dimensional FFTs faster. I use single CPU or GPU to calculate 1D FFT. It does not make any sense to parallelise a 1D FFT. But I came recently across the concept of out-of-core FFTs (seemingly parallel). I would like to know what everyone thinks about. Is it good idea to take a look at this? Let me know about your thoughts.

Thanks,
Arun

**Comment by navidcy** [comment]
hi @arunkv7, can you put a link or some more details about "out-of-core FFTs"? :)

**Reply by arunkv7** [reply]
Hi @navidcy ,
Here are some links regarding this:

1. https://pypi.org/project/out-of-core-fft/
2. https://link.springer.com/chapter/10.1007/978-1-4612-1516-5_14
3. https://github.com/moble/dask_fft
4. https://www.sciencedirect.com/science/article/pii/S0167819197001142

Arun

**Comment by glwagner** [comment]
@arunkv7 I don't think the out of core FFTs will provide speed up. Instead, they will allow you to compute the FFT when the array doesn't fit in memory. But for a 1D FFT this would have to be a very long vector indeed.
